lib/src/model/keychain.dart

```diff
@@ -257,25 +294,54 @@ KeyPair deriveArchethicKeypair(
   String curve = 'ed25519',
   String pathSuffix = '',
 }) {
-  // Hash the derivation path
-  final sha256 = Digest('SHA-256');
-  final hashedPath = sha256.process(
-    Uint8List.fromList(
-      replaceDerivationPathIndex(derivationPath, pathSuffix, index).codeUnits,
-    ),
+  final extendedSeed = deriveServiceSeed(
+    seed,
+    derivationPath,
+    index,
+    pathSuffix: pathSuffix,
   );
 
-  final hmac = crypto_lib.Hmac(crypto_lib.sha512, seed);
-  final digest = hmac.convert(hashedPath);
-  final extendedSeed = Uint8List.fromList(digest.bytes.sublist(0, 32));
-
   return crypto.generateDeterministicKeyPair(
     extendedSeed,
     curve,
     keychainOriginId,
   );
 }
 
+Uint8List deriveServiceSeed(
+  dynamic seed,
+  String derivationPath,
+  int index, {
+  String pathSuffix = '',
+}) {
+  final sha256 = Digest('SHA-256');
+  var hashedPath = Uint8List.fromList([]);
+  if (isPathWithIndex(derivationPath)) {
+    hashedPath = sha256.process(
+      Uint8List.fromList(
+        replaceDerivationPathIndex(derivationPath, pathSuffix, index).codeUnits,
+      ),
+    );
+  } else {
+    final path = derivationPath.split('/');
+    final serviceName = path.removeLast() + pathSuffix;
+    hashedPath = sha256.process(
```

Comment:
Is it normal that `sha256.process` takes a Uint8List instead of a string ?

Reply:
pointycastle provides this method like this. I never had any issue with this